guides/get-started-microservices-on-kubernetes/simple-microservice-llm-part1.adoc

```diff
@@ -0,0 +1,261 @@
+
+[source,bash,subs="normal"]
+----
+podman exec -it {ollama-pod-name} ollama pull {ollama-llm}
+----
+
+Test that your LLM is working by invoking its APIs like in the following:
+
+[source,bash,subs="normal"]
+----
+$ curl http://localhost:11434/api/generate -d '{ "model": "{ollama-llm}", "prompt":"Hi! My name is Tommaso"}'
+{"model":"llama3.1:8b","created_at":"2025-03-19T16:26:19.388567244Z","response":"C","done":false}
+{"model":"llama3.1:8b","created_at":"2025-03-19T16:26:19.518556766Z","response":"iao","done":false}
+{"model":"llama3.1:8b","created_at":"2025-03-19T16:26:19.646976123Z","response":" Tom","done":false}
+{"model":"llama3.1:8b","created_at":"2025-03-19T16:26:19.77417658Z","response":"mas","done":false}
+{"model":"llama3.1:8b","created_at":"2025-03-19T16:26:19.901190847Z","response":"o","done":false}
+{"model":"llama3.1:8b","created_at":"2025-03-19T16:26:20.033013914Z","response":"!","done":false}
+...
+----
+
+== Maven Project
+
+You will extend the sample application you created in link:simple-microservice-part1[{simple-microservice-part1}] by adding the link:https://github.com/wildfly-extras/wildfly-ai-feature-pack[wildfly-ai-feature-pack, window="_blank"].
+
+The link:https://github.com/wildfly-extras/wildfly-ai-feature-pack[wildfly-ai-feature-pack, window="_blank"] will:
+
+* add the necessary link:https://docs.langchain4j.dev[LangChain4J, window="_blank"] modules to the server (= the necessary dependencies to work with the model of choice)
+* add the configuration related to the model of choice; e.g. when using the `<layer>ollama-chat-model</layer>`, the feature pack would configure the server to connect to an external *Ollama* service
+
+=== pom.xml
+
+==== dependencies
+
+Add the following dependencies to the `pom-xml` file `dependencies` section:
+
+[source,xml,subs="normal"]
+----
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j</artifactId>
+            <version>{version-langchain4j}</version>
+            <scope>provided</scope>
+        </dependency>
+----
+
+NOTE: we are not adding LLM specific link:https://docs.langchain4j.dev[LangChain4J, window="_blank"] dependencies because they will be added automatically to the server by the `wildfly-ai-feature-pack` based on the specific AI layer we are using (e.g. `ollama-chat-model` will add the `dev.langchain4j:langchain4j-ollama` module)
+
+==== wildfly-maven-plugin
+
+Add the `wildfly-ai-feature-pack` feature-pack and the `ollama-chat-model` layer to the `wildfly-maven-plugin` configuration.
+
+You should end up with the link:https://github.com/wildfly/wildfly-maven-plugin/[wildfly-maven-plugin, window="_blank"] configured like in the following:
+
+[source,xml,subs="normal"]
+----
+    <plugin>
+        <groupId>org.wildfly.plugins</groupId>
+        <artifactId>wildfly-maven-plugin</artifactId>
+        <version>{version-wildfly-maven-plugin}</version>
```

Comment:
Hi, you could perhaps rely on the WildFly Glow discovery of feature-packs and layers?

Reply:
@jfdenise Thank you for the suggestion: if it's ok with you I'll add a note suggesting that Glow can be used ...
that's because I intentionally kept the example as simple as possible by not including e.g. Glow and not using any plugin to containerize the application or run it on k8s; that's because the whole series is intended to help closing the gap we have at the very beginning of our leaning path when I expect readers to just have basic understanding of Jakarta EE, maven and k8s;

Reply:
+1 with me it would be:
```
<discoverProvisioningInfo>
  <context>cloud</context>
  <spaces>
    <space>incubating</space>
  </spaces>
</discoverProvisioningInfo>
```

Reply:
Thank you @jfdenise, I added a note in the guide and a profile to [the](https://github.com/wildfly-extras/guides/pull/9) that show ho-to use Glow